GUAC-1182: Add swiss german keymap

spali opened the pull request.

mike-jumper commented:
Thanks for the contribution!

When you get a chance, please take a look at our [contribution guidelines](https://github.com/glyptodon/guacamole-client/blob/master/CONTRIBUTING). Everything generally looks fine here, but we do need a signed [CLA](http://glyptodon.org/cla.html) on file for all contributors. This is needed so we have a record of the legal right to merge and release your contribution (and of your right to contribute it).

spali commented:
I missed point (3) somehow. But to be honest, I was already scared to create a issue in Jira for such a small change. This explains why not more such smalls changes are contributed by others :(

I cloned an other keymap file and modified the makefile (both already included a copyright notice). With that I agreed to the license.

But because of that I can't sign the CLA because I don't know if the original author of the files did it and I don't have any possibility to prove that.

I know all the license stuff is important, but as soon as I don't contribute complete modules or some code I completely written by myself I don't understand why someone in general should sign such a CLA, because the license in the files should cover everything for such a case.
Also there is no reference in the license to this CLA.

So from my point of view:
When I modified the files I already accepted the license. With making the work public without any additional comment to the license, I allowed everyone to use the modifications as long as the original license is a respected.
So If you don't want to integrate my pull request without my sign on the CLA, you can just copy and paste my code (because I accepted your license already).
Then I'm not a contributor, just someone that has shared the work under the same license which allows you do take the code anyway by your self.

If it helps: I explain here, I resign any legal requirements or additional authorship  reference additional to the license included to the files in the code changes I made contained in this pull request.

Such things like insist on this CLA makes contribution for guys like me, which just try share some small improvements on a great piece of software for nothing, extremely dinky.

Take it or leave it.

mike-jumper commented:
> But to be honest, I was already scared to create a issue in Jira for such a small change. This explains why not more such smalls changes are contributed by others :(

JIRA won't bite. Each change needs a corresponding issue so we can schedule the internal time to review the change, merge, test, etc. There's more at work here than the effort involved in the coding of the contribution itself, even if the contribution seems trivial.

> When I modified the files I already accepted the license.

Unfortunately, that's not how the license is written. You can make modifications to MIT licensed code, and your modifications need not be under the same license. The GPL works the way you describe, but we chose to use the MIT license for exactly that reason.

> Such things like insist on this CLA makes contribution for guys like me, which just try share some small improvements on a great piece of software for nothing, extremely dinky.

I do feel your pain ... but to be honest, to me this sounds like: "you have airbags, seat belts, and crumple zones - no wonder you don't make many cars". One signature per contributor is, IMHO, a small price to pay for the safety of the project and the community that uses it.

We did have this discussion internally when deciding to adopt a CLA and contribution guidelines, neither of which existed until a few months ago. After much research and reading of opinions, we decided that being able to legally defend the project was necessary for its future.

[The Wikipedia article on CLAs](http://en.wikipedia.org/wiki/Contributor_License_Agreement) has a good paragraph on all this:

> The purpose of a CLA is to ensure that the guardian of a project's outputs has the necessary ownership or grants of rights over all contributions to allow them to distribute under the chosen licence. In some cases this will mean that the contributor will assign the copyright in all contributions to the project owner; in other cases, they will grant an irrevocable licence to allow the project maintainer to use the contribution. CLAs also have roles in raising awareness of IPR issues within a project.

The CLA is not a testament that you know the rest of our codebase is properly licensed (though it is), nor even a statement that you are licensing your work under our license. It only states that you are giving us the right to release your contribution. If your contribution is based on our existing code, your contribution is your modifications, not the pieces that we already have.

There's a [great discussion about all this on StackExchange](http://programmers.stackexchange.com/questions/168020/how-signing-out-a-cla-prevents-legal-issues-in-open-source-projects), if interested.

If you change your mind, let us know, and the PR will be reopened.
